tests/treemodel/node.js

```diff
@@ -178,4 +179,58 @@ describe( 'Node', () => {
 			} );
 		} );
 	} );
+
+	describe( 'toJSON', () => {
+		it( 'should serialize empty node', () => {
+			let node = new Element( 'one' );
+
+			expect( treeModelTestUtils.jsonParseStringify( node ) ).to.deep.equal( {
+				_attrs: [],
+				_children: {
+					_indexMap: [],
+					_nodes: []
+				},
+				name: 'one',
+				parent: null
+			} );
+		} );
+
+		it( 'should serialize node with attributes', () => {
+			let node = new Element( 'one', { foo: true, bar: false } );
+
+			expect( treeModelTestUtils.jsonParseStringify( node ) ).to.deep.equal( {
+				_attrs: [ [ 'foo', true ], [ 'bar', false ] ],
+				_children: {
+					_indexMap: [],
+					_nodes: []
+				},
+				name: 'one',
+				parent: null
+			} );
+		} );
+
+		it( 'should serialize node with children', () => {
+			expect( treeModelTestUtils.jsonParseStringify( root ) ).to.deep.equal( {
+				_attrs: [],
+				_children: {
+					_indexMap: [ 0, 1, 2 ],
+					_nodes: [
+						{ _attrs: [], _children: { _indexMap: [], _nodes: [] }, name: 'one', parent: null },
+						{
+							_attrs: [], _children: {
+							_indexMap: [ 0, 0, 1, 2 ], _nodes: [
+								{ _attrs: [], text: 'ba', parent: 'two' },
+								{ _attrs: [], _children: { _indexMap: [], _nodes: [] }, name: 'img', parent: 'two' },
+								{ _attrs: [], text: 'r', parent: 'two' }
+							]
+						}, name: 'two', parent: null
+						},
```

Comment:
This object isn't correctly formatted. Two `}` at the same level of indentation mean that something is wrong here.